File: frontend/frontend/src/app/models/category.ts
import { IBookList } from "./book";

export class Category{
    id : 0;
    name: string;
    description : string;
    book : IBookList[] = [];
    active : boolean;

    constructor(){
        this.id = 0;
        this.name = "";
        this.description = "";
        this.active = true;
    }
}

export interface IListCategory{
    id : 0;
    name: string;
    description : string;
    book : IBookList[];
    active : boolean;
}

File: backend/BookstoreTamAn/src/main/java/com/minhnghia/datn/BookstoreTamAn/model/Category.java
package com.minhnghia.datn.BookstoreTamAn.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Entity
@Table(name = "category")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Category extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;
    private String description;
    private String createdBy;
    private String modifyBy;

    @ManyToMany(mappedBy = "categories", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Book> books;
}
